packages/color-modes/src/index.tsx

```diff
@@ -146,17 +147,38 @@ const applyColorMode = (theme: Theme, mode: string | undefined): Theme => {
   })
 }
 
+const assembleColorModes = (theme: Theme, outer: Theme) => {
+  const { colors = {}, initialColorModeName } = outer
+
+  const mutatedColors = get(theme, 'colors', {})
+  const modes = get(outer, 'colors.modes', {})
+
+  if (!('modes' in colors)) return theme.colors
+
+  return merge.all({}, mutatedColors, {
+    modes: {
+      [initialColorModeName || '__default']: omit(colors, ['modes']),
+      ...modes,
+    },
+  })
+}
+
 export const ColorModeProvider: React.FC = ({ children }) => {
   const outer = useThemeUI()
   const [colorMode, setColorMode] = useColorModeState(outer.theme)
 
   const theme = applyColorMode(outer.theme || {}, colorMode)
+
   if (theme.useCustomProperties !== false) {
     // TODO: This mutation is less than ideal
     // We could save custom properties to `theme.colorVars`,
     // But it's infeasible to do this because of how the packages are split.
-    theme.rawColors = theme.colors
-    theme.colors = toCustomProperties(theme.colors, 'colors')
+
+    theme.rawColors = assembleColorModes(theme, outer.theme)
+    theme.colors = toCustomProperties(
+      omit(outer.theme.colors, ['modes']),
+      'colors'
+    )
```

Comment:
A little digression:

What do you guys think about wrapping this entire block from line 170 to line 182 in useMemo? 
This entire logic doesn't need to happen on each render, it could be just computed whenever the mode changes.

Reply:
Love it!